### PR TITLE
[5.8] Fix FormRequest validate not twice.

### PR DESCRIPTION
**Pull request by ttsuru**
Add tests to #26419
Fix validate method.

refs #25736

### CODE REVIEW

File: src/Illuminate/Foundation/Http/FormRequest.php [hunk 1]
```diff
@@ -172,7 +181,7 @@ protected function failedAuthorization()
      */
     public function validated()
     {
-        return $this->getValidatorInstance()->validated();
+        return $this->validator->validated();
```
Comment (driesvints) [inline src/Illuminate/Foundation/Http/FormRequest.php]: What if you call this method before the validator is set? I think we should think this through more.

Comment (ttsuru) [reply src/Illuminate/Foundation/Http/FormRequest.php]: @driesvints 
Thsnks for checking.
How about adding `ValidatesWhenResolvedTrait::validateResolved()` ?

Comment (ttsuru) [reply src/Illuminate/Foundation/Http/FormRequest.php]: @driesvints 
Thanks for checking.
`ValidatesWhenResolvedTrait::validateResolved()` is called in `FormRequestServiceProvider::boot()`.

So I think that it is normally set.

Comment (mohammedmanssour) [reply src/Illuminate/Foundation/Http/FormRequest.php]: can't this be fixed by adding a simple if statement that can be added to the `getValidatorInstance`?

Comment (ttsuru) [reply src/Illuminate/Foundation/Http/FormRequest.php]: @mohamed-aiman 
I think so too, but getValidatorInstance called twice.
We require single validator instance.

**Pull request merged**